AsyncDisplayKit/ASCellNode.m

```diff
@@ -106,21 +105,12 @@ - (instancetype)init
   return self;
 }
 
-- (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
+- (id<ASLayoutable>)layoutSpecThatFits:(ASSizeRange)constrainedSize
```

Comment:
Is there a use case where we would want this method to return something other than ASLayoutSpec itself?  I see the cases where an ASLayoutSpec would want to directly contain an ASDisplayNode, rather than wrapping with the ASCompositeNode/Spec.  But especially because the name of this method is layoutSpecThatFits:, it's a bit confusing to me if it should actually allow returning an ASDisplayNode with nothing else.  I suspect the answer is no...but that we perhaps should support returning an ASLayout as well as ASLayoutSpec, so if you wanted to manually create an ASLayout and set it up to describe the position of subnodes, you could do that without subclassing ASLayoutSpec.

Thanks for any thoughts on this.  Since this API will be very commonly used and indeed is arguably /the most important API/ which ASDK adds on top of UIKit, it's very important that we nail the naming and semantics.  I think we are very close and the new ASLayoutSpec name makes a great deal of sense, but let's think carefully about any related method names and their return types.